api/handler_readme.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/skeswa/gophr/common/errors"
+)
+
+const (
+	queryStringRepoTextKey = "r"
+	queryStringRefTextKey  = "hb"
+)
+
+var (
+	repoName string
+	ref      string
+)
+
+// ReadmeHandler creates an HTTP request handler that responds to fuzzy package
+// searches.
+func ReadmeHandler() func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		// Get URL Query Params
+		qs := r.URL.Query()
+		if qs != nil {
+			// Get repo name
+			repoName = qs.Get(queryStringRepoTextKey)
```

Comment:
I'd like to point out this route is a clone of `handler_search.go`, are you saying that handler_search would suffer from the same problems you're mentioning above and would require a refactor?

```
package main

import (
	"net/http"

	"github.com/gocql/gocql"
	"github.com/skeswa/gophr/common/dtos"
	"github.com/skeswa/gophr/common/errors"
	"github.com/skeswa/gophr/common/models"
)

const (
	queryStringSearchTextKey = "q"
)

// SearchHandler creates an HTTP request handler that responds to fuzzy package
// searches.
func SearchHandler(session *gocql.Session) func(http.ResponseWriter, *http.Request) {
	return func(w http.ResponseWriter, r *http.Request) {
		var searchText string

		qs := r.URL.Query()
		if qs != nil {
			searchText = qs.Get(queryStringSearchTextKey)
		}

		if len(searchText) < 1 {
			errors.RespondWithError(w, NewInvalidQueryStringParameterError(
				queryStringSearchTextKey,
				searchText,
			))
			return
		}

		packageModels, err := models.FuzzySearchPackages(session, searchText)
		if err != nil {
			errors.RespondWithError(w, err)
			return
		}

		packageListDTO := dtos.NewPackageListDTOFromPackageModelList(packageModels)
		if len(packageListDTO) > 0 {
			json, err := packageListDTO.MarshalJSON()
			if err != nil {
				errors.RespondWithError(w, err)
				return
			}

			w.WriteHeader(http.StatusOK)
			w.Write(json)
		} else {
			w.WriteHeader(http.StatusOK)
			w.Write([]byte(dtos.EmptyListJSON))
		}
	}
}
```